src/pkg/services/m365/m365.go

```diff
@@ -72,6 +81,28 @@ func Users(ctx context.Context, acct account.Account, errs *fault.Bus) ([]*User,
 			return nil, clues.Wrap(err, "formatting user data")
 		}
 
+		userInfo, err := discovery.UserDetails(ctx, acct, pu.ID, errs)
+		if err != nil {
+			return nil, clues.Wrap(err, "getting user details")
+		}
+
+		pu.HasMailBox = userInfo.HasMailBox
+		pu.HasOnedrive = userInfo.HasOneDrive
+
+		// mailbox and onedrive info is independent of mailboxSetting.
+		// so getting them and then checking if any error getting mailboxSettings
+		if userInfo.ErrGetMailBoxSetting != "" {
+			pu.ErrGettingUserInfo = userInfo.ErrGetMailBoxSetting
+			continue
+		}
```

Comment:
This conditional is more effort than necessary.  We have no nil-pointer issues to be concerned with, and golang zero-values are viable will suffice for all the other values.  Easier overall just to plug in the value and not worry about the rest.
```suggestion
		pu.ErrGettingUserInfo = userInfo.ErrGetMailBoxSetting
```

Although, replacing this with a userInfo struct is going to be the easiest way forward overall.

Reply:
Actually, it fails if the mailbox settings values are blanks.

Reply:
done